qcodes/instrument_drivers/american_magnetics/AMI430.py

```diff
@@ -0,0 +1,636 @@
+import logging
+import numpy as np
+import time
+
+from qcodes import Instrument, VisaInstrument
+from qcodes.utils.validators import Numbers
+
+
+def R_y(theta):
+    """ Construct rotation matrix around y-axis. """
+    return np.array([[np.cos(theta), 0, np.sin(theta)],
+                     [0, 1, 0],
+                     [-np.sin(theta), 0, np.cos(theta)]])
+
+
+def R_z(theta):
+    """ Construct rotation matrix around z-axis. """
+    return np.array([[np.cos(theta), -np.sin(theta), 0],
+                     [np.sin(theta), np.cos(theta), 0],
+                     [0, 0, 1]])
+
+
+class AMI430(VisaInstrument):
+    """
+    Driver for the American Magnetics Model 430 magnet power supply programmer.
+
+    This class controls a single magnet power supply. In order to use two or
+    three magnets simultaniously to set field vectors, first instantiate the
+    individual magnets using this class and then pass them as arguments to
+    either the AMI430_2D or AMI430_3D virtual instrument classes.
+
+    Args:
+        name (string): a name for the instrument
+        address (string): IP address of the power supply programmer
+        coil_constant (float): coil constant in Tesla per ampere
+        current_rating (float): maximum current rating in ampere
+        current_ramp_limit (float): current ramp limit in ampere per second
+        persistent_switch (bool): whether this magnet has a persistent switch
+    """
+    def __init__(self, name, address, coil_constant, current_rating,
+                 current_ramp_limit, persistent_switch=True,
+                 reset=False, **kwargs):
+        super().__init__(name, address, terminator='\n', **kwargs)
+
+        self._parent_instrument = None
+
+        self._coil_constant = coil_constant
+        self._current_rating = current_rating
+        self._current_ramp_limit = current_ramp_limit
+        self._persistent_switch = persistent_switch
+
+        self._field_rating = coil_constant * current_rating
+        self._field_ramp_limit = coil_constant * current_ramp_limit
+
+        # Make sure the ramp rate time unit is seconds
+        if self.ask('RAMP:RATE:UNITS?') == '1':
```

Comment:
You're right that it can be figured out, but maybe it would be good to have the code a little bit more resilient to changes of the terminator. Some check that the answer is either ("0" or "1") instead of only checking =="1". If something would go wrong with this temrinator, the user will not find out I think if there is no check. If the magnet returns "0\r\n" suddenly, then this will leave the magnet with the wrong units. So that would be hard to debug